src/v/bytes/tests/iobuf_bench.cc

```diff
@@ -28,8 +30,38 @@ size_t move_bench() {
     perf_tests::stop_measuring_time();
     return inner_iters * 2;
 }
+
+// This microbench mocks common pattern in Redpanda where smaller iobufs will be
+// parsed out of a larger iobuf then appendded together. This pattern is heavily
+// used in the datalake impl.
+template<int Bufs, size_t BufSize>
+size_t append_bench() {
+    static_assert(Bufs > 1, "test requires more than one buffer to append");
+
+    iobuf_parser parser(random_generators::make_iobuf(Bufs * BufSize));
+
+    std::vector<iobuf> buffers;
+    buffers.reserve(Bufs);
+    for (int i = 0; i < Bufs; i++) {
+        buffers.push_back(parser.share(BufSize));
+    }
+
+    iobuf& res = buffers[0];
```

Comment:
Maybe just `iobuf res;` then append all the bufs instead of skipping the first, etc? Simplifies and close to what `encode_plain` does I think.

Reply:
Fair point, updated it as you suggested.